library/ajax/template_context_search.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Ajax interface for custom template context.
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2019 Jerry Padgett <sjpadgett@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+$cq = <<< createQuery
+CREATE OR REPLACE VIEW zView_user_templates AS
+SELECT cl2.cl_list_slno context_id, cl2.cl_list_item_long context_text,
+cl3.cl_list_slno cat_id, cl3.cl_list_item_long cat_text, cl3.cl_creator cat_ownerid,
+IFNULL(cl4.cl_list_slno,"") tmpl_id, IFNULL(cl4.cl_list_item_long,"") tmpl_text, IFNULL(cl4.cl_creator,"") tmpl_ownerid
+FROM customlists cl2
+INNER JOIN customlists cl3 on cl2.cl_list_slno=cl3.cl_list_id
+LEFT OUTER JOIN customlists cl4 on cl3.cl_list_slno=cl4.cl_list_id and cl4.cl_list_type=4 and cl4.cl_deleted=0
+WHERE cl2.cl_list_type=2 and cl2.cl_deleted=0
+and cl3.cl_list_type=3 and cl3.cl_deleted=0
+GROUP BY context_id;
+createQuery;
+
+require_once(dirname(__FILE__) . '/../../interface/globals.php');
+
+if (!verifyCsrfToken($_GET["csrf_token_form"])) {
+    csrfNotVerified();
+}
+
+$search = $_GET['search'];
+$eSearch = "%" . $search . "%";
+$results = [];
+$error = sqlStatementNoLog($cq);
+$sq = "Select `context_id` as `id`, `context_text` as `text` From `zView_user_templates` Where `context_text` Like ?";
+
+$r = sqlStatementNoLog($sq, array($eSearch));
```

Comment:
Having been there and done more 'z' views than standard tables, pl consider `SHOW TABLES` check before this DDL.

I hear every DBAs in the world coming after me for writing this.  Sorry.

Reply:
we could give each view a randomly created uuid :)